dips/dip-158.md

```diff
@@ -0,0 +1,805 @@
+  "command_type": "CaptureCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "CaptureCommand",      
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98"
+  }
+}
+
+```
+#### CaptureCommand Response
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a"    
+}
+```
+
+### VoidCommand
+| Command Type |Direction| Description | Request Data | Response Data |
+|-|-|-|-|-|
+|VoidCommand|Receiver > Sender|This functionality allows the receiver to void the payment. For example, the merchant voided the payment because the purchased products are out of stock. Relevant only for Auth / Capture payments|Reference ID ; Receiver Status|Sender Status|
+
+#### VoidCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|status|[PaymentStatusObject](#PaymentStatusObject)|Y|Receiver status. For the void command the status should be abort and the abort code should be void|
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "VoidCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "VoidCommand",
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+    "status": {
+      "status": "abort",
+      "abort_code": "void",
+      "abort_message": "abort message"
+    }
+  }
+}
+```
+#### VoidCommand Response
+Field|Type|Required?|Description|
+|-|-|-|-|
+|status|[PaymentStatusObject](#PaymentStatusObject)|Y|Sender status. Used to verify that the sender VASP acknoledges the void command by responding with the abort status|
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a",
+  "result": {
+    "_ObjectType": "PaymentStatusObject",
+    "status": {
+      "status": "abort"
+    }
+  }
+}
+```
+### AbortCommand
+| Command Type |Direction| Description | Request Data | Response Data |
+|-|-|-|-|-|
+|AbortCommand|Sender > Receiver / Receiver > Sender|This functionality allows both parties to abort the payment. For example, the receiver may want to abort the payment following failed risk checks|Reference ID ; Requester (Sender / Receiver) Status|Responder (Sender / Receiver) Status| 
+
+
+#### AbortCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|status|[PaymentStatusObject](#PaymentStatusObject)|Y|Initiator status (can be either the sender or receiver). For the abort command the status should be abort|
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "AbortCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "AbortCommand",    
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+    "status": {
+      "status":"abort",
```

Comment:
similarly, what if I send an abort but set the status to something else? let's define some of the edge cases.